### PR TITLE
Enhance User Guidance Docs: Clarify When to Use Mason

### PR DESCRIPTION
**Pull request by bmcdonald3**
This pull request introduces new documentation that clarifies when to leverage Mason for building Chapel applications and when alternative approaches might be more suitable.

The new documentation provides a breakdown of ideal use cases for Mason, including:

- Pure Chapel Applications: Mason excels at managing the build process for projects written entirely in Chapel.
- External Dependencies with Spack or pkg-config: Mason integrates seamlessly with external dependencies managed by these popular package managers.

Additionally, the documentation highlights scenarios where alternative build tools like Make might be better suited, such as:

- Complex Compilation Needs: When a Chapel program requires intricate build commands beyond the capabilities of the Chapel compiler, tools like Make offer greater flexibility.

This improved guidance empowers users to make informed decisions about which build tool best aligns with their specific project requirements.

### CODE REVIEW

File: doc/rst/tools/mason/mason.rst [hunk 1]
```diff
@@ -13,17 +13,39 @@ that be to create a library to contribute to the community, use an
...
-   branch <https://chapel-lang.org/docs/main/mason-packages/>`_ of the
-   documentation (which tracks real-time GitHub development), you may
-   find that it contains a more complete set of sections.  We welcome
-   feedback and requests on the guide as we go.
-
+When to Leverage Mason for Chapel Builds
+----------------------------------------
+
+Mason simplifies the build process for Chapel applications, but it's
+not a one-size-fits-all solution. Here's a breakdown of when Mason
```
Comment (benharsh) [inline doc/rst/tools/mason/mason.rst:20]: "not yet a ...", or "not currently"?

File: doc/rst/tools/mason/mason.rst [hunk 2]
```diff
@@ -13,17 +13,38 @@ that be to create a library to contribute to the community, use an
...
-   branch <https://chapel-lang.org/docs/main/mason-packages/>`_ of the
-   documentation (which tracks real-time GitHub development), you may
-   find that it contains a more complete set of sections.  We welcome
-   feedback and requests on the guide as we go.
-
+When to Leverage Mason for Chapel Builds
+----------------------------------------
+
+Mason simplifies the build process for Chapel applications, but it's
+not currently a one-size-fits-all solution. Here's a breakdown of
```
Comment (mstrout) [inline doc/rst/tools/mason/mason.rst:20]: I recommend deleting this last phrase, "but it's not currently ...".  The next sentence covers the key point that there are possible alternatives.  This phrase might come off as defensive.

File: doc/rst/tools/mason/mason.rst [hunk 3]
```diff
@@ -13,17 +13,38 @@ that be to create a library to contribute to the community, use an
...
+   `pkg-config`. If your project uses such dependencies, Mason can handle
+   them efficiently.
+
+**Considering Alternatives:**
+
+1. Complex Compilation Needs Beyond `chpl`: For Chapel programs requiring
+   compilation steps beyond what the Chapel compiler offers, Mason might not
+   be the most suitable choice. Consider alternatives like `Make` for greater
+   flexibility in crafting intricate build commands. As an example, the
+   Arkouda project, a prominent open-source Chapel application, utilizes
```
Comment (mstrout) [inline doc/rst/tools/mason/mason.rst:40]: "utilizes" --> "uses"

**Review by mstrout — COMMENTED** [review]
Looking good!  Just a couple of minor wording suggestions.

**Pull request merged**